src/app/shared/directives/destroyable.directive.ts

```diff
@@ -0,0 +1,12 @@
+import { Directive, OnDestroy } from '@angular/core';
+import { Subject } from 'rxjs';
+
+@Directive()
+export class DestroyableDirective implements OnDestroy {
+  protected readonly destroy$: Subject<void> = new Subject();
+
+  public ngOnDestroy(): void {
```

Comment:
I think piblic is redundant, it's public by default

Reply:
but I like the idea! Can you create a refactoring issue to implement such an approach all over the project?